client/src/js/users/components/__tests__/Detail.test.js

```diff
@@ -0,0 +1,137 @@
+import { UserDetail, mapStateToProps, mapDispatchToProps } from "../Detail";
+
+describe("<UserDetail />", () => {
+    let props;
+
+    beforeEach(() => {
+        props = {
+            onGetUser: jest.fn(),
+            match: {
+                params: {
+                    userId: "foo"
+                }
+            },
+            groupsFetched: true,
+            onListGroups: jest.fn(),
+            onSetUserRole: jest.fn(),
+            detail: {
+                id: "foo",
+                groups: "bar",
+                administrator: "baz",
+                identicon: "foo",
+                primary_group: "Bar",
+                permissions: { foo: "bar" }
+            },
+            onRemoveUser: jest.fn(),
+            error: []
+        };
+    });
+
+    it("should render", () => {
+        const wrapper = shallow(<UserDetail {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it("componentDidMount should call onGetUser", () => {
+        expect(props.onGetUser).not.toHaveBeenCalled();
+        const wrapper = shallow(<UserDetail {...props} />);
+        expect(props.onGetUser).toHaveBeenCalledWith("foo");
+    });
+
+    it("componentDidMount should call onListGroups", () => {
+        const wrapper = shallow(<UserDetail {...props} />);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['wrapper' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/virtool/virtool/pullRequest?prid=4503767)